File: ApiHelper/Common/GetOptions.cs
﻿using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace RestApi.Common
{
    public class GetOptions
    {
        [Range(1, int.MaxValue)]
        public int Page { get; set; } = 1;

        [Range(0, int.MaxValue)]
        public int? RowsCount { get; set; }

        public IEnumerable<SortOption> Sort { get; set; }

        public IEnumerable<Filter> Filters { get; set; }
    }
}

File: ApiHelper/ClientApp/Common/GetOptions.js
//@flow
import SortOption from "./SortOption";

export default class GetOptions {
    page: number;
    rowsCount: ?number;
    sort: ?SortOption[];

    constructor(page: number, rowsCount: ?number, sort: ?SortOption[]) {
        this.page = page;
        this.rowsCount = rowsCount;
        this.sort = sort;
    }
}